pinot-core/src/main/java/com/linkedin/pinot/core/periodictask/PeriodicTaskScheduler.java

```diff
@@ -15,58 +15,63 @@
  */
 package com.linkedin.pinot.core.periodictask;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+
 /**
  * Periodic task scheduler will schedule a list of tasks based on their initial delay time and interval time.
  */
 public class PeriodicTaskScheduler {
   private static final Logger LOGGER = LoggerFactory.getLogger(PeriodicTaskScheduler.class);
-  private static final int CORE_POOL_SIZE = 5;
-  private final ScheduledExecutorService _executorService;
 
-  public PeriodicTaskScheduler() {
-    LOGGER.info("Initializing PeriodicTaskScheduler.");
-    _executorService = Executors.newScheduledThreadPool(CORE_POOL_SIZE);
-  }
+  private ScheduledExecutorService _executorService;
 
   /**
    * Start scheduling periodic tasks.
    */
   public void start(List<PeriodicTask> periodicTasks) {
-    if (periodicTasks == null || periodicTasks.isEmpty()) {
-      LOGGER.warn("No periodic task assigned to scheduler!");
-      return;
+    if (_executorService != null) {
+      LOGGER.warn("Periodic task scheduler already started");
     }
 
-    if (periodicTasks.size() > CORE_POOL_SIZE) {
-      LOGGER.warn("The number of tasks:{} is more than the default number of threads:{}.", periodicTasks.size(),
-          CORE_POOL_SIZE);
+    List<PeriodicTask> tasksWithValidInterval = new ArrayList<>();
+    for (PeriodicTask periodicTask : periodicTasks) {
+      if (periodicTask.getIntervalInSeconds() > 0) {
+        tasksWithValidInterval.add(periodicTask);
+      }
```

Comment:
Would be good to add a log whether or not you are adding a periodic task here (and remove other logs of "Adding XXX task")

Reply:
Done.